src/client/mod.rs

```diff
@@ -137,48 +189,71 @@ impl InfluxDbClient {
         };
 
         let any_value = q as &dyn Any;
+        let basic_parameters: Vec<(String, String)> = (self.clone()).into();
```

Comment:
Is there a better option than cloning?

Reply:
We could change the into impl to 
```rust
impl<'a> Into<Vec<(String, String)>> for &'a InfluxDbClient {
    fn into(self) ->  Vec<(String, String)> {
        let mut vec: Vec<(String, String)> = Vec::new();
        vec.push(("db".to_string(), self.database.to_owned()));
        if let Some(auth) = &self.auth {
            vec.push(("u".to_string(), auth.username.to_owned()));
            vec.push(("p".to_string(), auth.password.to_owned()));
        }
        vec
    }
}
```

Reply:
I added this one as a second impl so that &InfluxDbClient and InfluxDbClient can be used.